### ISSUE TITLE
A simulcast transceiver saved from rollback by addTrack doesn't re-associate, but unicast does

### ISSUE THREAD

**Issue by jan-ivar** [w3c/webrtc-pc#2796]
A [JSEP rollback quirk](https://www.rfc-editor.org/rfc/rfc8829.html#section-5.7) makes it possible to save 🎣 a transceiver from the 🦈 jaws of rollback by associating it with `addTrack`:
_"... an RtpTransceiver MUST NOT be removed if a track was attached to the RtpTransceiver via the addTrack method. This is so that an application may call addTrack, then call setRemoteDescription with an offer, then roll back that offer, then call createOffer and have an "m=" section for the added track appear in the generated offer."_

When I [test this](https://jsfiddle.net/jib1/0qf8uc3b/), I find some interesting results (Chrome and Safari):
> PASS transceiver saved from rollback of sRD(simulcastOffer) by addTrack is simulcast
FAIL transceiver saved from rollback of sRD(simulcastOffer) by addTrack can be re-associated: Expected exactly one transceiver - Expected 1, got 2 failed
PASS transceiver saved from rollback of sRD(unicastOffer) by addTrack can be re-associated

Basically, if you were looking for a way to add simulcast encodings with addTrack, then yay (who needs addTransceiver?) 🙃

But I also found that a simulcast transceiver saved from rollback by addTrack doesn't re-associate, whereas a unicast transceiver does.

This doesn't seem POLA. Not sure what I expected, but it seems odd that they behave differently perhaps. What should it do? 🤷🏼‍♂️

**Comment by jan-ivar** [comment]
The above tests used this order: sRD(simulcastOffer), addTrack, rollback.

Naively, I expected the same result as addTrack, sRD(simulcastOffer), rollback, but [it differs](https://jsfiddle.net/jib1/0qf8uc3b/51) (in Chrome):
> PASS transceiver saved from rollback of sRD(simulcastOffer) by upfront addTrack is **_unicast_**
FAIL transceiver saved from rollback of sRD(simulcastOffer) by upfront addTrack can be re-associated: Expected exactly one transceiver - Expected 1, got 2 failed
PASS transceiver saved from rollback of sRD(unicastOffer) by upfront addTrack can be re-associated

(and Safari):
> FAIL transceiver saved from rollback of sRD(simulcastOffer) by upfront addTrack is **_unicast_**: Expected exactly one transceiver - Expected 1, got 2 failed
FAIL transceiver saved from rollback of sRD(simulcastOffer) by upfront addTrack can be re-associated: Expected exactly one transceiver - Expected 1, got 2 failed
PASS transceiver saved from rollback of sRD(unicastOffer) by upfront addTrack can be re-associated

The main takeaway (from Chrome) is simulcast is rolled back to unicast for transceivers created by addTrack, but not for transceivers created by sRD.

I think the solution is to roll back simulcast to unicast for transceivers created by sRD as well, if they're going to stick around.

This seems a defensible interpretation of [RFC 8829 (section 4.1.10.2)](https://www.rfc-editor.org/rfc/rfc8829.html#section-4.1.10.2) which says rollback _"discards any proposed changes to the session, returning the state machine to the "stable" state"_.

**Comment by docfaraday** [comment]
I concur that the reasonable thing to do in this situation is to roll back to ridless unicast.

**Comment by jan-ivar** [comment]
Solved by https://github.com/w3c/webrtc-pc/pull/2797.
